cmd/clusterctl/examples/openstack/generate-yaml.sh

```diff
@@ -164,11 +165,23 @@ tenant-id=\"$PROJECT_ID\"
 domain-name=\"$DOMAIN_NAME\"
 "
 
+if [ "$CACERT_ORIGINAL" != "null" ]; then
+  OPENSTACK_CLOUD_PROVIDER_CONF_PLAIN="$OPENSTACK_CLOUD_PROVIDER_CONF_PLAIN
+  ca-file=\"$CACERT\"
+  "
+fi
+
 OS=$(uname)
 if [[ "$OS" =~ "Linux" ]]; then
   OPENSTACK_CLOUD_PROVIDER_CONF=$(echo "$OPENSTACK_CLOUD_PROVIDER_CONF_PLAIN"|base64 -w0)
+  if [ "$CACERT_ORIGINAL" != "null" ]; then
+    OPENSTACK_CLOUD_CACERT_CONFIG=$(cat "$CACERT_ORIGINAL"|base64 -w0)
```

Comment:
the $CACERT_ORIGINAL was the value of cacert in clouds.yaml I think, so may be we should use `echo` but not `cat` here.

refer the test result below:
```
[root@hchenxa1 openstack]#
[root@hchenxa1 openstack]# OPENSTACK_CLOUD_CONFIG_PLAIN=$(cat "clouds.yaml")
[root@hchenxa1 openstack]# CLOUD=openstack
[root@hchenxa1 openstack]# CACERT_ORIGINAL=$(echo "$OPENSTACK_CLOUD_CONFIG_PLAIN" | yq r - clouds.$CLOUD.cacert)
[root@hchenxa1 openstack]# echo $CACERT_ORIGINAL
-----BEGIN CERTIFICATE----- MIID5TCCAs2gAwIBAgIBFDANBgkqhkiG9w0BAQsFADBiMQswCQYDVQQGEwJVUzE0 MDIGA1UEChMrSW50ZXJuYXRpb25hbCBCdXNpbmVzcyBNYWNoaW5lcyBDb3Jwb3Jh dGlvbjEdMBsGA1UEAxMUSUJNIEludGVybmFsIFJvb3QgQ0EwHhcNMTYwMjI0MDUw MDAwWhcNMzUwMTAzMDQ1OTU5WjBiMQswCQYDVQQGEwJVUzE0MDIGA1UEChMrSW50 ZXJuYXRpb25hbCBCdXNpbmVzcyBNYWNoaW5lcyBDb3Jwb3JhdGlvbjEdMBsGA1UE AxMUSUJNIEludGVybmFsIFJvb3QgQ0EwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAw ggEKAoIBAQDUKGuk9Tmri43R3SauS7gY9rQ9DXvRwklnbW+3Ts8/Meb4MPPxezdE cqVJtHVc3kinDpzVMeKJXlB8CABBpxMBSLApmIQywEKoVd0H0w62Yc3rYuhv03iY y6OozBV0BL6tzZE0UbvtLGuAQXMZ7ehzxqIta85JjfFN86AO2u7xrNF0FYyGH+E0 Rn6yNhb25VrqxE0OYbSMIGoWdvS11K4SgVDqrJ9OqIk8NHrIJ8Ed24P/YPMeAp3j U409Gev1zGcuLdRr09WckQ145FZVDbPq42gcl7qYICPhZ4/eDUUjFgxpipfMGkMb 1X+Y3kFDgb4BO8Xrdda2VQo1iDZs8A8bAgMBAAGjgaUwgaIwPwYJYIZIAYb4QgEN BDIWMEdlbmVyYXRlZCBieSB0aGUgU2VjdXJpdHkgU2VydmVyIGZvciB6L09TIChS QUNGKTAOBgNVHQ8BAf8EBAMCAQYwDwYDVR0TAQH/BAUwAwEB/zAdBgNVHQ4EFgQU +d4Y5Z4wE2lRp/15hUiMfA5v2OMwHwYDVR0jBBgwFoAU+d4Y5Z4wE2lRp/15hUiM fA5v2OMwDQYJKoZIhvcNAQELBQADggEBAH87Ms8yFyAb9nXesaKjTHksLi1VKe2i zESWozYFXnRtOgOW7/0xXcfK+7PW6xwcOqvTk61fqTGxj+iRyZf2e3FNtIB+T/Lg 3SZF9sztPM0jEUELWycC8l6WPTvzQjZZBCsF+cWbU1nxvRNQluzCsTDUEIfThJIF cLu0WkoQclUrC3d2tM8jclLPssb6/OV8GaJ+4mx4ri7HbGaUAOtA/TXKR6AuhgkR NPKYhpPU0q/PRlGXdwJP8zXb8+CXMMTnI5Upur7Tc5T3I/x1Gqfz7n1sTRZfsuiQ J5uua4hz4te3oV2tm7LWcNItHD43zttBTTx/m5icg71JE2gcr2oincw= -----END CERTIFICATE-----
[root@hchenxa1 openstack]# cat $CACERT_ORIGINAL
cat: unrecognized option '-----BEGIN'
Try 'cat --help' for more information.
```

Reply:
@hchenxa Besides this issue, does there are any other issues that we need to raise with this PR?

@hidekazuna can you help update?

Reply:
I am ok now, I prefer to do some offline test before merge (http and https) ... thoughts?

Reply:
Yes, I think @hidekazuna still need some update for his PR.

Reply:
`$CACERT_ORIGINAL` is the path of CA file. Could you provide your executed command? I tested the following.
```
./generate-yaml.sh --provider-os ubuntu -c /home/ubuntu/.config/openstack/clouds.yaml
```

Reply:
The commands example should like below:
```
[root@hchenxa1 openstack]# ./generate-yaml.sh  -h
generate-yaml.sh - generates a provider-configs.yaml file

Usage:
generate-yaml.sh [options] <path/to/clouds.yaml> <cloud> <provider os: [centos,ubuntu,coreos]>
options:
-h, --help                    show brief help
-f, --force-overwrite         if file to be generated already exists, force script to overwrite it
```

and I run the command like below:
```
[root@hchenxa1 openstack]# ./generate-yaml.sh clouds.yaml openstack centos
```

Reply:
@hchenxa can you help try

```
./generate-yaml.sh -f ./clouds.yaml openstack ubuntu
```

Reply:
I found missed cmd/clusterctl/examples/openstack/provider-component/user-data/centos/templates/master-user-data.sh and worker-user-data.sh. I will update.

Reply:
I updated by https://github.com/kubernetes-sigs/cluster-api-provider-openstack/commit/0bf8a1bf20fa24ef22c193d48b2f5bf292527979

Reply:
@hchenxa you should configure the `cacert` as the path to a CA Cert, but not the content of the cert.

```
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/vendor/github.com/gophercloud/utils/openstack/clientconfig/results.go#L33-L35
```

Reply:
sure